Formula/dtm.rb

```diff
@@ -0,0 +1,17 @@
+class Dtm < Formula
+  desc "Cross-language distributed transaction manager"
+  homepage "http://d.dtm.pub"
+  url "https://github.com/dtm-labs/dtm/archive/refs/tags/v1.8.1.tar.gz"
+  sha256 "64fae905308fd3ab6e5e0152aa98e388e37eb73764af82b5c6212121ff96892c"
+  license "BSD-3-Clause"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-o", bin/"dtm"
+  end
+
+  test do
+    assert_equal "version: 0.0.0-dev commit: NA built at: NA\n", shell_output("#{bin}/dtm -v")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.